feat(lsp/fmt): support ignore files & patterns

Sysix opened the pull request:
closes https://github.com/oxc-project/oxc/issues/15459

> This PR adds support for ignore files (.gitignore, .prettierignore) and ignore patterns (ignorePatterns in config) to the language server formatter

The server does not support a custom ignore path, it is the same for the server linter and its CLI integration too :)

Sysix commented:
* **#15647** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15647?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15647?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15647).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

leaysgur reviewed (approved):
(I'm not familiar with LS but, ) looks good to me.

camc314 commented:
### Merge activity

* **Nov 14, 11:27 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 14, 11:27 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
feat(lsp/fmt): support ignore files & patterns (#15647)

closes https://github.com/oxc-project/oxc/issues/15459

> This PR adds support for ignore files (.gitignore, .prettierignore) and ignore patterns (ignorePatterns in config) to the language server formatter

The server does not support a custom ignore path, it is the same for the server linter and its CLI integration too :)
